handel/src/services/stepfunctions/index.ts

```diff
@@ -0,0 +1,138 @@
+    return deletePhasesCommon.unDeployService(ownServiceContext, SERVICE_NAME);
+}
+
+export const producedEventsSupportedServices = [];
+
+export const producedDeployOutputTypes = ['environmentVariables', 'policies'];
+
+export const consumedDeployOutputTypes = ['environmentVariables', 'policies'];
+
+export const supportsTagging = false;
+
+function generateDefinitionString(filename: string, dependenciesDeployContexts: DeployContext[]): string {
+    const readFile = path.extname(filename) == '.json' ? util.readJsonFileSync : util.readYamlFileSync;
+    const definitionFile = readFile(filename);
+    const dependencyArns: Map<string, string> = new Map();
+    // Map service name to ARN
+    for (const context of dependenciesDeployContexts) {
+        dependencyArns.set(context.serviceName, context.eventOutputs.lambdaArn);
+    }
+    // Change 'resource' in each state from service name to ARN
+    for (const state_name in definitionFile.States) {
+        const state = definitionFile.States[state_name];
+        if (definitionFile.States.hasOwnProperty(state_name) && 'Resource' in state) {
+            state.Resource = dependencyArns.get(state.Resource);
+        }
+    }
+    return JSON.stringify(definitionFile);
+}
+
+function getCompiledStepFunctionsTemplate(stackName: string, ownServiceContext: ServiceContext<StepFunctionsConfig>, dependenciesDeployContexts: DeployContext[]): Promise<string> {
+    const definitionString = generateDefinitionString(ownServiceContext.params.definition, dependenciesDeployContexts);
+    const policyStatements = deployPhaseCommon.getAllPolicyStatementsForServiceRole([], dependenciesDeployContexts);
+    const handlebarsParams: HandlebarsStepFunctionsTemplate = {
+        stateMachineName: stackName,
+        definitionString,
+        policyStatements
+    };
+    return handlebarsUtils.compileTemplate(`${__dirname}/stepfunctions-template.yml`, handlebarsParams);
+}
+
+function getDeployContext(serviceContext: ServiceContext<StepFunctionsConfig>, cfStack: AWS.CloudFormation.Stack): DeployContext {
+    const deployContext = new DeployContext(serviceContext);
+    const stateMachineArn = cloudFormationCalls.getOutput('StateMachineArn', cfStack);
+    const stateMachineName = cloudFormationCalls.getOutput('StateMachineName', cfStack);
+    // Output policy for consuming this state machine
+    deployContext.policies.push({
+        'Effect': 'Allow',
+        'Action': [
+            'states:StartExecution',
+            'states:StopExecution'
+        ],
+        'Resource': [
+            stateMachineArn
+        ]
+    });
+
+    // Inject env vars
+    deployContext.addEnvironmentVariables(deployPhaseCommon.getInjectedEnvVarsFor(serviceContext, {
+        STATE_MACHINE_ARN: stateMachineArn,
+        STATE_MACHINE_NAME: stateMachineName
+    }));
+
+    // Inject event outputs
+    deployContext.eventOutputs.stateMachineArn = stateMachineArn;
```

Comment:
You can probably remove these few lines setting eventOutputs since this service won't produce or consume events from other services.